llvm/lib/Transforms/Vectorize/SandboxVectorizer/DependencyGraph.cpp

```diff
@@ -50,29 +51,137 @@ MemDGNodeIntervalBuilder::make(const Interval<Instruction> &Instrs,
                              cast<MemDGNode>(DAG.getNode(MemBotI)));
 }
 
+DependencyGraph::DependencyType
+DependencyGraph::getRoughDepType(Instruction *FromI, Instruction *ToI) {
+  // TODO: Perhaps compile-time improvement by skipping if neither is mem?
+  if (FromI->mayWriteToMemory()) {
+    if (ToI->mayReadFromMemory())
+      return DependencyType::RAW;
```

Comment:
is RAW more conservative than WAW?

Reply:
They are about the same in the sense that you have to force ordering for all RAW, WAW or WAR.